### PR TITLE
Add XLA:GPU Benchmarking Workflow

### PR DESCRIPTION
**Pull request by mariecwhite**

### CODE REVIEW

File: iree-tf/benchmark/benchmark_model.py [hunk 1]
```diff
@@ -28,241 +29,229 @@
...
-        return ("RESNET50", resnet50.ResNet50, model_definition.input_batch_size)
-    elif unique_id.startswith(unique_ids.MODEL_BERT_LARGE_FP32_TF):
-        return ("BERT_LARGE", bert_large.BertLarge, model_definition.input_batch_size)
-    elif unique_id.startswith(unique_ids.MODEL_T5_LARGE_FP32_TF):
-        return ("T5_LARGE", t5_large.T5Large, model_definition.input_batch_size)
-    else:
-        raise ValueError(f"Model definition not supported")
+  model_definition = tf_model_definitions.TF_MODELS_DICT[unique_id]
+  if unique_id.startswith(unique_ids.MODEL_RESNET50_FP32_TF):
+    return ("RESNET50", resnet50.ResNet50, model_definition)
```
Comment (pzread) [inline iree-tf/benchmark/benchmark_model.py:37]: (No need to address here): IIUC here we still download the (latest?) models from huggingface. Maybe we can load from the saved models in the ModelArtifact?

File: oobi/benchmark-definitions/python/data_types.py [hunk 2]
```diff
@@ -48,12 +73,141 @@ class Model(object):
...
   source_info: str
   input_batch_size: int
   inputs: ModelData
   outputs: ModelData
+  # A list of artifacts derived from this model.
+  artifacts: List[ModelArtifact]
 
   def __str__(self):
     return self.name
+
```
Comment (pzread) [inline oobi/benchmark-definitions/python/data_types.py:87]: It looks like the classes below are not used right now. I'm still prototyping them. Maybe we can remove them now and I'll add back later.

**Review by pzread — APPROVED** [review]

File: .github/workflows/benchmark_xla_gpu.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+# Benchmarks XLA:GPU workloads.
+
+name: Benchmark XLA:GPU
+
+on:
+  schedule:
+    # Scheduled to run at 09:00 UTC and 21:00 UTC.
+    - cron: '0 09,21 * * *'
+  workflow_dispatch:
+  # TODO: Remove.
```
Comment (pzread) [inline .github/workflows/benchmark_xla_gpu.yml:16]: Reminder of the cleanup

File: iree-tf/benchmark/benchmark_model.py (hunk 1 above)
Comment (mariecwhite) [reply iree-tf/benchmark/benchmark_model.py:37]: Good point. We may want to freeze the huggingface and keras versions in the pip requirements.txt file. I think loading the SavedModels will only load the frozen weights and not necessarily any tweaks made to the model architecture. Let's put an action item to handle this - could be fixing dependencies to a specific version/sha, creating a docker image with the venv already setup, etc.

File: oobi/benchmark-definitions/python/data_types.py (hunk 2 above)
Comment (mariecwhite) [reply oobi/benchmark-definitions/python/data_types.py:87]: Removed

File: .github/workflows/benchmark_xla_gpu.yml (hunk 3 above)
Comment (mariecwhite) [reply .github/workflows/benchmark_xla_gpu.yml]: Removed

**Pull request merged**